server/src/routes/user.ts

```diff
@@ -125,12 +114,73 @@ user.get("/me", authenticateUser, async (c) => {
     })
     .from(users)
     .where(eq(users.email, email));
-
   if (!user) {
     return c.json({ success: false, error: "User not found" }, 404);
   }
-
   return c.json({ success: true, data: { user } });
 });
 
+user.patch("/", authenticateUser, zValidator("json", UpdateUserSchema), async (c) => {
+  const { email: currentEmail } = c.get("jwtPayload");
+  const updateData = c.req.valid("json");
+
+  try {
+    const [updatedUser] = await db
+      .update(users)
+      .set({
+        ...updateData,
+        updatedAt: new Date(),
```

Comment:
`updatedAt` will automatically get updated on update? if that is the case whats the necessity to update `updatedAt`?